### PR TITLE
oidc authenticator: allow http.Client to be overridden

### PR DESCRIPTION
**Pull request by enj**
This change allows the http.Client used by the OIDC authenticator to
be overridden.  This is useful when this code is being used as a
library outside of core Kubernetes.  For example, a downstream
consumer may want to override the http.Client's internals such as
its TLS configuration.

Signed-off-by: Monis Khan <mok@vmware.com>

/kind cleanup
/milestone v1.23
/triage accepted
/priority backlog
/assign @liggitt

xref: #99765

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go [hunk 1]
```diff
@@ -81,9 +81,12 @@ type Options struct {
 	// See: https://openid.net/specs/openid-connect-core-1_0.html#IDToken
 	ClientID string
 
-	// PEM encoded root certificate contents of the provider.
+	// PEM encoded root certificate contents of the provider.  Ignored when Client is not nil.
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go:84]: would it be better to make this and `Client` mutually exclusive?

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go [hunk 2]
```diff
@@ -527,6 +527,49 @@ func TestToken(t *testing.T) {
...
 			},
 		},
+		{
+			name: "groups-distributed invalid client",
+			options: Options{
+				IssuerURL:     "{{.URL}}",
+				ClientID:      "my-client",
+				UsernameClaim: "username",
+				GroupsClaim:   "groups",
+				Client:        http.DefaultClient, // cause distributed claims fetching to fail with an unknown CA error
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go:537]: any way to verify the error or verify this client gets called?

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go (hunk 1 above)
Comment (enj) [reply staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go]: Yeah I think so.  Updated and added unit test.

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go (hunk 2 above)
Comment (enj) [reply staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go]: Updated the test logic so that I could assert the returned error.

**Pull request merged**